### PR TITLE
list: add --include-progress and --include-workspace-size

### LINKED ISSUE #510: cli: more performance optimisations for `list` command
The speed up of the `list` command was improved as part of https://github.com/reanahub/reana-client/issues/367. However, more improvements are needed.  

E.g. for one client having about 101 workflows, the `list` may take 40-50 seconds, which is too much:

```console
$ python -m cProfile -s cumulative /afs/cern.ch/user/r/reana/public/reana-qa/bin/reana-client list 
...
      128   44.979    0.351   44.979    0.351 {method 'read' of '_ssl._SSLSocket' objects}
        1    0.000    0.000   44.976   44.976 client.py:1302(getresponse)
        1    0.000    0.000   44.976   44.976 client.py:300(begin)
        1    0.000    0.000   44.973   44.973 client.py:267(_read_status)
       18    0.000    0.000   44.973    2.498 {method 'readline' of '_io.BufferedReader' objects}
```

We should debug what is being done on the server side, what is being sent to the client, and minimise passing of any unnecessary information down to the client so that `reana-client list` would be very fast.

Ideally one SQL query should be done on the server side, detecting and passing only that information that is being displayed:

```console
$ reana-client list
NAME              RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS
bsm09             2            2021-04-05T18:12:31   2021-04-05T18:12:32   2021-04-05T18:22:59   finished
bsm08             2            2021-04-05T18:12:29   2021-04-05T18:12:30   2021-04-05T18:22:56   finished
bsm07             2            2021-04-05T18:12:27   2021-04-05T18:12:28   2021-04-05T18:44:58   finished
bsm06             2            2021-04-05T18:12:25   2021-04-05T18:12:26   2021-04-05T18:22:11   finished
bsm05             2            2021-04-05T18:12:22   2021-04-05T18:12:23   2021-04-05T18:23:08   finished
...
```

i.e. only information about times and statuses, which should be very fast and should not take dozens of seconds.

P.S. The command `list` took 44 secs to get response, the command `list -v` took 56 seconds.

### PR DESCRIPTION
**Pull request by ParthS007**
Screenshot 2021-05-11 at 4 40 32 PM

performance optimisations for list command

closes #510

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ Version 0.8.0 (UNRELEASED)
 - Adds job start and finish times to ``logs`` command output.
 - Changes ``delete`` to prevent workflow hard deletion.
 - Changes ``list`` to display interactive session status.
+- Adds ``--include-progress`` and ``--include-workspace-size`` in ``list`` command.
```
Comment (mvidalgarcia) [inline CHANGES.rst:12]: Could you please be more explicit? As a user, I might not know what these two flags refer to. I would expect some minimum explanation on why those flags were introduced.

```
- Changes ``list`` command to stop including workflow progress and workspace size by default. Adds ``--include-progress`` and ``--include-workspace-size`` options to add them to the output.

File: tests/test_cli_workflows.py [hunk 2]
```diff
@@ -68,7 +72,9 @@ def test_workflows_server_ok():
             "reana_client.api.client.current_rs_api_client",
             make_mock_api_client("reana-server")(mock_response, mock_http_response),
         ):
-            result = runner.invoke(cli, ["list", "-t", reana_token])
```
Comment (mvidalgarcia) [inline tests/test_cli_workflows.py:71]: Could you add another test case that covers the call without `--include-progress`, and maybe also another for `--include-workspace-size`?

File: reana_client/cli/workflow.py [hunk 3]
```diff
@@ -111,6 +111,20 @@ def workflow_execution_group(ctx):
...
+    default=False,
+    help="Include progress information of the workflows.",
+)
+@click.option(
+    "--include-workspace-size",
+    "include_workspace_size",
+    is_flag=True,
+    default=False,
+    help="Include size information of the workspace.",
+)
```
Comment (mvidalgarcia) [inline reana_client/cli/workflow.py:127]: I think we can set the defaults to `None` here, so we avoid sending these as query params all the time.

**Review by mvidalgarcia — COMMENTED** [review]
I'm failing to understand the purpose of the new flags from the user point of view.
```console
$ reana-client list                                                                                                                                 
NAME                             RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
root6-roofit-cwl-kubernetes      1            2021-05-10T12:55:04   2021-05-10T12:55:09   2021-05-10T12:55:23   finished
root6-roofit-serial-kubernetes   1            2021-05-10T12:54:28   2021-05-10T12:54:42   2021-05-10T12:54:55   finished
root6-roofit-yadage-kubernetes   1            2021-05-10T12:53:44   2021-05-10T12:53:49   2021-05-10T12:54:18   finished

$ reana-client list --include-progress                                                                                                              
NAME                             RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
root6-roofit-cwl-kubernetes      1            2021-05-10T12:55:04   2021-05-10T12:55:09   2021-05-10T12:55:23   finished
root6-roofit-serial-kubernetes   1            2021-05-10T12:54:28   2021-05-10T12:54:42   2021-05-10T12:54:55   finished
root6-roofit-yadage-kubernetes   1            2021-05-10T12:53:44   2021-05-10T12:53:49   2021-05-10T12:54:18   finished

$ reana-client list --include-workspace-size
NAME                             RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
root6-roofit-cwl-kubernetes      1            2021-05-10T12:55:04   2021-05-10T12:55:09   2021-05-10T12:55:23   finished
root6-roofit-serial-kubernetes   1            2021-05-10T12:54:28   2021-05-10T12:54:42   2021-05-10T12:54:55   finished
root6-roofit-yadage-kubernetes   1            2021-05-10T12:53:44   2021-05-10T12:53:49   2021-05-10T12:54:18   finished

$ reana-client list --include-progress --include-workspace-size
NAME                             RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
root6-roofit-cwl-kubernetes      1            2021-05-10T12:55:04   2021-05-10T12:55:09   2021-05-10T12:55:23   finished
root6-roofit-serial-kubernetes   1            2021-05-10T12:54:28   2021-05-10T12:54:42   2021-05-10T12:54:55   finished
root6-roofit-yadage-kubernetes   1            2021-05-10T12:53:44   2021-05-10T12:53:49   2021-05-10T12:54:18   finished

```
Am I missing something? cc/ @tiborsimko

**Review by mvidalgarcia — COMMENTED** [review]
`reana-client status` fails for me, can you reproduce?

```console
$ reana-client status -w hello  
Cannot retrieve the status of a workflow {'created': '2021-05-17T10:17:49', 'id': '72d92d12-717e-4b97-aa9d-17357c72cccb', 'logs': '{"12087630-5e76-4be2-8bbc-556c19d0b950": {"workflow_uuid": "72d92d12-717e-4b97-aa9d-17357c72cccb", "job_name": "", "compute_backend": "Kubernetes", "backend_job_id": "reana-run-job-e5aefa06-cced-4de3-9847-5e2f399fc732", "docker_img": "python:2.7-slim", "cmd": "python \\"code/helloworld.py\\" --inputfile \\"data/names.txt\\" --outputfile \\"results/greetings.txt\\" --sleeptime 0", "status": "finished", "logs": "job: :\\n 2021-05-17T10:18:02 INFO     Parameters: inputfile=data/names.txt outputfile=results/greetings.txt sleeptime=0.0\\n\\n\\nCompleted\\n", "started_at": "2021-05-17T10:18:01", "finished_at": "2021-05-17T10:18:08"}}', 'name': 'hello.2', 'progress': {'run_finished_at': '2021-05-17T10:18:08', 'run_started_at': '2021-05-17T10:18:01'}, 'status': 'finished', 'user': '00000000-0000-0000-0000-000000000000'}: 
'total'
```

**Comment by ParthS007** [comment]
@mvidalgarcia I checked with all `r-client` commands, working fine. 

I pushed a change in `r-w-controller` which will fix the `status` error. 👍

**Review by mvidalgarcia — APPROVED** [review]

**Pull request merged**